Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/QueryInfos.java

```diff
@@ -107,7 +117,13 @@ public void setFullScopeFound(boolean fullScopeFound) {
 	public String toQueryString(Context context, TypeQueryTerm typeQuery, GeoQueryTerm geoQuery,
 			LanguageQueryTerm langQuery, boolean ignoredId) {
 		StringBuilder result = new StringBuilder("?");
-
+		if(qQueryTerm!=null){
+			result.append("q=")
+					.append(qQueryTerm.getAttribute())
+					.append(qQueryTerm.getOperator())
+					.append(qQueryTerm.getOperant())
+					.append('&');
+		}
```

Comment:
this will only work on the first entry of a query and doesn't follow to next of qQueryTerm. But we shouldn't filter here anyway as the qQuery might only be true after merging. We need to filter according to q after the merge of the entities. Have a look at the calculate function of QQueryTerm we need something similar for this to run after the final merge just directly on the Map<String, Object> entries of the entities not on the BaseProperty classes that the sub man uses.